pkgs/by-name/se/segger-systemview/package.nix

```
@@ -0,0 +1,188 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, segger-jlink
+, config
+, acceptLicense ? config.segger-systemview.acceptLicense or false
```

```suggestion
, acceptLicense ? config.segger-systemview.acceptLicense or false
```

Please remove it. It makes no sense at all.

---

Why does it not make sense?

For the segger software, the license acceptance check happens at the time of download, not when starting the program. We have the same system for both android and j-link.

---

See also https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/by-name/se/segger-jlink/package.nix#L41 and https://www.segger.com/purchase/licensing/ , the same license agreements presumably apply to SystemView.

---

This type of interaction is not ideal. Relying on configuration in `config` is even less ideal. This is not how a normal package manager should work. If emphasis on license is very important, just use a custom unfree license with link in `meta.license`.

---

I agree with you that it's not ideal. I would really like to find an alternative solution if possible.

I was going to make a long comment about this, but I realise that it is only valid for j-link. Systemview seems to have a more standard proprietary license, requiring you to agree before you use the software, not before you download. Let's set `meta.license = licenses.unfree` on this one.

---

The original problem was that we click an "accept license" button for the user when we download j-link, with a data field in a POST request. This license states that the user needs to accept it before even being allowed to download the software. This is different from many other proprietary programs, where the download is irrelevant, but you have to accept the license before you use it.

We cannot click this button for the user if we are not absolutely sure that they have accepted the license, because we would be the ones breaking the license in place of the user. Setting a custom license on the package is not a guarantee that the user has actually accepted it. For other software, when the user has downloaded the software and choose not to read the license, then that's not really our problem. We have not checked any boxes, nor accepted anything by downloading the binary. But for j-link, this is our problem.

At least this was the original reasoning, but I'd be happy to do an 180 if it turns out we can either do this in a smarter way, or if it turns out that the license does not hold these implications.

---

Thanks for pointing this out, SystemView indeed has no pre-download license agreement. I have updates the PR accordingly, now no extra confirmation is required.